docs/using.md

```diff
@@ -0,0 +1,282 @@
+
+For example, consider the following problematic code:
+
+```python
+...
+
+gForceRunningTotal = 0.0
+
+def on_dataframe_received_handler(stream_consumer: qx.StreamConsumer, df: pd.DataFrame):
+    print(df)
+    # Calculate gForceTotal, the sum of vector absolute values 
+    df["gForceTotal"] = df["gForceX"].abs() + df["gForceY"].abs() + df["gForceZ"].abs() 
+    
+    # Track running total of all g-forces
+    global gForceRunningTotal
+    gForceRunningTotal += df["gForceTotal"]
+
+    # write result data to output topic
+    topic_producer.get_or_create_stream(stream_consumer.stream_id).timeseries.publish(df)
+...
+```
+
+You might think this would give you the running total for a stream, but because the callback is registered for each stream, you'd actually get all streams modifying the global.
+
+If you were running across multiple replicas, you'd get a running total for each replica, because each replica would have its own instance of the global variable. Again, the results would not be as you might expect.
+
+Let's say there were three streams and two replicas, you'd get the running total of two streams for one replica, and the running total for the third stream in the other replica.
+
+In most practical scenarios you'd want to track a running total per stream (say, total g-forces per race car), or perhaps for some variables a running total across all streams. Each of these scenarios is described in the followng sections.
+
+## Tracking running totals per stream
+
+Sometimes you might want to calculate a running total of a variable for a stream. For example, the total g-force a racing car is exposed to. If you use a global variable you'll lose the stream context. All streams will add to the value potentially, and each replica will also have its own instance of the global, further confusing matters.
+
+The solution is to use the stream context to preserve a running total for that stream only. To do this you can use the `stream_id` of a stream to identify its data. Consider the following example:
+
+```python
+...
+g_running_total_per_stream = {}
+
+def callback_handler (stream_consumer: qx.StreamConsumer, data: qx.TimeseriesData):
+
+    if stream_consumer.stream_id not in g_running_total_per_stream:
+        g_running_total_per_stream[stream_consumer.stream_id] = 0
+    
+    ...
+
+    g_running_total_per_stream[stream_consumer.stream_id] += some_value
+
+...
+```
+
+The key point here is that data is tracked per stream context. You keep running totals on a per-stream basis by using the stream ID, `stream_consumer.stream_id` to index a dictionary containing running totals for each stream.
+
+!!! note
+
+    A stream will only ever be processed by one replica.
+
+## Tracking running totals across multiple streams
```

Comment:
Yep. I like this section. The user could achieve this in multiple ways but using QuixStreams is a valid option.
We could write a `cache app` example implementation